src/modules/forms/EndpointForm/EndpointForm.js

```diff
@@ -175,7 +177,7 @@ class EndpointForm extends PureComponent {
             key='download'
             mod='primary'
             type='button'
-            onClick={() => downloadObjectAsJson(this.props.api, this.props.api.name)}
+            onClick={() => downloadObjectAsJson(getUpdatedEndpoint(this.props.api), this.props.api.name)}
```

Comment:
You could call `getUpdatedEndpoint(this.props.api)` once and use it in both places